scripts/init.groovy.d/aa_configure_credentials.groovy

```diff
@@ -0,0 +1,370 @@
+// output init hook stage
+println '### JENKINS AUTOCONFIG: CREDENTIALS ###'
+
+/**
+  * Check if Jenkins should not configure itself and skip.
+  */
+
+Boolean skipConfiguringJenkins() {
+    !(System.env?.ENVIRONMENT in ['dev', 'staging']) ||
+        (new File("${Jenkins.instance.rootDir}/autoConfigComplete").exists())
+}
+if(skipConfiguringJenkins()) {
+    println 'Skipping Jenkins auto-configuration'
+    return
+}
+
+
+/**
+  * From this point onward Jenkins should configure itself.
+  */
+import jenkins.*
+import jenkins.model.*
+import hudson.*
+import hudson.model.*
+
+import org.yaml.snakeyaml.LoaderOptions
+import org.yaml.snakeyaml.Yaml
+import org.yaml.snakeyaml.constructor.SafeConstructor
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse
+
+import com.cloudbees.jenkins.plugins.sshcredentials.impl.BasicSSHUserPrivateKey
+import com.cloudbees.plugins.credentials.CredentialsMatchers
+import com.cloudbees.plugins.credentials.CredentialsProvider
+import com.cloudbees.plugins.credentials.CredentialsScope
+import com.cloudbees.plugins.credentials.SystemCredentialsProvider
+import com.cloudbees.plugins.credentials.domains.Domain
+import com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl
+import hudson.security.ACL
+import hudson.util.Secret
+import org.jenkinsci.plugins.plaincredentials.impl.StringCredentialsImpl
+
+Map getSecret(Map options) {
+    String secretName = options.secretId
+    Region region = Region.of(options.region)
+    SecretsManagerClient client = SecretsManagerClient.builder()
+        .region(region)
+        .build()
+    GetSecretValueRequest getSecretValueRequest = GetSecretValueRequest.builder()
+        .secretId(secretName)
+        .build()
+    GetSecretValueResponse getSecretValueResponse = client.getSecretValue(getSecretValueRequest)
+    String secret = getSecretValueResponse.secretString()
+    LoaderOptions loaderOptions = new LoaderOptions()
+    def yaml = new Yaml(new SafeConstructor(loaderOptions))
```

Comment:
```suggestion
    Yaml yaml = new Yaml(new SafeConstructor(loaderOptions))
```

Reply:
https://bitbucket.org/snakeyaml/snakeyaml/wiki/Documentation (I think the recommended instantiation pattern changed awhile back after they fixed some Snakeyaml vulnerability)

Reply:
implemented